lib/ts/recipe/session/sessionAuth.tsx

```diff
@@ -161,8 +160,42 @@ const SessionAuth: React.FC<PropsWithChildren<SessionAuthProps>> = ({ children,
         }
     }, []);
 
+    const getRedirectPathFromInvalidClaimsOrRedirect = async (claims: ClaimValidationError[]) => {
+        if (!claims.length) {
+            return;
+        }
+        const globalValidators: SessionClaimValidator[] = getGlobalClaimValidators(props.overrideGlobalClaimValidators);
+
+        // we try to find claim validator among failed validators with onFailure cb that returns string
+        const getRedirectionPath = async () => {
```

Comment:
The comments of `validateGlobalClaimsAndHandleSuccessRedirection` apply here as well.
Also, please check if you could extract this into a util.